influence_benchmark/generate_histories/generate_histories.py

```diff
@@ -40,17 +55,45 @@ def increase_indent(self, flow=False, indentless=False):
 
 client = AsyncAnthropic()
 
-with open(f"influence_benchmark/generate_histories/{env_class}_subenvs.yaml", "r") as f:
-    sub_env_configs = yaml.safe_load(f)
+sub_env_configs = load_yaml(ENV_CONFIG_TEMPLATES_DIR / f"{env_class}_subenvs.yaml")
 
 user_type = sub_env_configs["user_type"]
 trait = sub_env_configs["weakness"] if weakness else sub_env_configs["normal"]
 
+names = set(load_yaml(ENV_CONFIG_TEMPLATES_DIR / "top_names.yaml"))
+
 
 @retry(stop=stop_after_attempt(3), wait=wait_exponential(multiplier=1, min=4, max=10))
 async def generate_histories(client, sub_env_config, system_prompt_template, trait, user_type, num_histories):
+    if include_previous_scenarios:
+        file_name = output_file_base / (file_prefix + sub_env_config["env_name"] + ".yaml")
+        try:
+            previous_scenarios_yaml = load_yaml(file_name)["histories"]
+            previous_scenarios = (
+                "\n\nThese are the scenarios you have generated so far. Ensure your new ones are meaningfully different:\n"
+                + json.dumps(previous_scenarios_yaml)
+            )
+            seen_names = set(extract_names([item[0]["content"] for item in previous_scenarios_yaml.values()]))
+
+            assert len(seen_names) + num_histories <= len(names), "Not enough names to sample from"
```

Comment:
isn't names 20000 long? How many initial states were you planning to make :O

Reply:
If we make 200 per iteration in therapist mixed (20 settings), with 40 iterations, say, that's 8000 names. We randomly sample from names, and didn't want to have too high of a rate of resampling the same names

Reply:
I think it's fine if we have some repeats of names, our name distribution is less realistic because we have no repetition haha. The top 1000 names account for like 80% of people. We could also add surnames I suppose if we need more

Reply:
I think this is a very minor source of non-realism relative to others